eras/shelley/test-suite/test/Test/Control/Iterate/SplitMapRules.hs

```diff
@@ -0,0 +1,256 @@
+t05 xs v = (Dom (RExclude (Base SplitR xs) (SetSingleton v)))
+
+t06 :: SplitMap SS Int -> SS -> Exp (Sett SS ())
+t06 xs v = (Dom (DRestrict (SetSingleton v) (Base SplitR xs)))
+
+t07 :: SplitMap SS Int -> Sett SS () -> Exp (Sett SS ())
+t07 xs set = (Dom (DRestrict (Base SetR set) (Base SplitR xs)))
+
+t08 :: SplitMap SS Int -> SS -> Exp (Sett SS ())
+t08 xs v = (Dom (DExclude (SetSingleton v) (Base SplitR xs)))
+
+t09 :: SplitMap SS Int -> Sett SS () -> Exp (Sett SS ())
+t09 xs set = (Dom (DExclude (Base SetR set) (Base SplitR xs)))
+
+t10 :: SplitMap SS Int -> Sett SS () -> Exp (SplitMap SS Int)
+t10 xs set = (DRestrict (Base SetR set) (Base SplitR xs))
+
+t11 :: SplitMap SS Int -> SS -> Exp (SplitMap SS Int)
+t11 xs k = (DRestrict (SetSingleton k) (Base SplitR xs))
+
+t12 :: SplitMap SS Int -> SS -> () -> Exp (SplitMap SS Int)
+t12 xs k v = (DRestrict (Singleton k v) (Base SplitR xs))
+
+t13 :: SplitMap SS Char -> Map SS Int -> Exp (SplitMap SS Char)
+t13 xs m = (DRestrict (Dom (Base MapR m)) (Base SplitR xs))
+
+t14 :: SplitMap SS Int -> Map SS Char -> Exp (Map SS Char)
+t14 xs m = (DRestrict (Dom (Base SplitR xs)) (Base MapR m))
+
+t15 :: SplitMap SS Int -> SplitMap SS Char -> Exp (SplitMap SS Char)
+t15 xs zs = (DRestrict (Dom (Base SplitR xs)) (Base SplitR zs))
+
+t16 :: SplitMap SS Char -> Map SS Int -> Int -> Exp (SplitMap SS Char)
+t16 xs m k = (DRestrict (Dom (RRestrict (Base MapR m) (SetSingleton k))) (Base SplitR xs))
+
+t17 :: SplitMap SS Char -> Map SS Int -> Map Int () -> Exp (SplitMap SS Char)
+t17 xs m1 m2 = (DRestrict (Dom (RRestrict (Base MapR m1) (Base MapR m2))) (Base SplitR xs))
+
+t18 :: SplitMap SS Int -> Map SS () -> Exp (SplitMap SS Int)
+t18 xs set = (DRestrict (Base MapR set) (Base SplitR xs))
+
+t19 :: SplitMap SS Int -> SS -> Exp (SplitMap SS Int)
+t19 xs k = (DExclude (SetSingleton k) (Base SplitR xs))
+
+t20 :: SplitMap SS Int -> SS -> Char -> Exp (SplitMap SS Int)
+t20 xs k v = (DExclude (Dom (Singleton k v)) (Base SplitR xs))
+
+t21 :: SplitMap SS Int -> Int -> SS -> Exp (SplitMap SS Int)
+t21 xs k v = (DExclude (Rng (Singleton k v)) (Base SplitR xs))
+
+t22 :: SplitMap SS Int -> Set.Set SS -> Exp (SplitMap SS Int)
+t22 xs set = (DExclude (Base SetR (Sett set)) (Base SplitR xs))
+
+t23 :: SplitMap SS Int -> Map SS Char -> Exp (SplitMap SS Int)
+t23 xs m = (DExclude (Dom (Base MapR m)) (Base SplitR xs))
+
+t24 :: SplitMap SS Int -> SplitMap SS Char -> Exp (SplitMap SS Char)
+t24 xs zs = (DExclude (Dom (Base SplitR xs)) (Base SplitR zs))
+
+t25 :: SplitMap SS Int -> Int -> Exp (SplitMap SS Int)
+t25 xs u = (RExclude (Base SplitR xs) (SetSingleton u))
+
+t26 :: SplitMap SS Int -> Sett Int () -> Exp (SplitMap SS Int)
+t26 xs set = (RExclude (Base SplitR xs) (Base SetR set))
+
+t27 :: SplitMap SS Int -> Int -> Exp (SplitMap SS Int)
+t27 xs k = (RRestrict (Base SplitR xs) (SetSingleton k))
+
+t28 :: SplitMap SS Int -> Set.Set Int -> Exp (SplitMap SS Int)
+t28 xs s = (RRestrict (Base SplitR xs) (Base SetR (Sett s)))
```

Comment:
Redundant parenthesis:
```suggestion
t01 :: SplitMap SS Int -> Exp (Sett SS ())
t01 xs = Dom (Base SplitR xs)

t02 :: SplitMap SS Int -> Int -> Exp (Sett SS ())
t02 xs v = Dom (RRestrict (Base SplitR xs) (SetSingleton v))

t03 :: SplitMap SS Int -> Sett Int () -> Exp (Sett SS ())
t03 xs set = Dom (RRestrict (Base SplitR xs) (Base SetR set))

t04 :: SplitMap SS Int -> Sett Int () -> Exp (Sett SS ())
t04 xs set = Dom (RExclude (Base SplitR xs) (Base SetR set))

t05 :: SplitMap SS Int -> Int -> Exp (Sett SS ())
t05 xs v = Dom (RExclude (Base SplitR xs) (SetSingleton v))

t06 :: SplitMap SS Int -> SS -> Exp (Sett SS ())
t06 xs v = Dom (DRestrict (SetSingleton v) (Base SplitR xs))

t07 :: SplitMap SS Int -> Sett SS () -> Exp (Sett SS ())
t07 xs set = Dom (DRestrict (Base SetR set) (Base SplitR xs))

t08 :: SplitMap SS Int -> SS -> Exp (Sett SS ())
t08 xs v = Dom (DExclude (SetSingleton v) (Base SplitR xs))

t09 :: SplitMap SS Int -> Sett SS () -> Exp (Sett SS ())
t09 xs set = Dom (DExclude (Base SetR set) (Base SplitR xs))

t10 :: SplitMap SS Int -> Sett SS () -> Exp (SplitMap SS Int)
t10 xs set = DRestrict (Base SetR set) (Base SplitR xs)

t11 :: SplitMap SS Int -> SS -> Exp (SplitMap SS Int)
t11 xs k = DRestrict (SetSingleton k) (Base SplitR xs)

t12 :: SplitMap SS Int -> SS -> () -> Exp (SplitMap SS Int)
t12 xs k v = DRestrict (Singleton k v) (Base SplitR xs)

t13 :: SplitMap SS Char -> Map SS Int -> Exp (SplitMap SS Char)
t13 xs m = DRestrict (Dom (Base MapR m)) (Base SplitR xs)

t14 :: SplitMap SS Int -> Map SS Char -> Exp (Map SS Char)
t14 xs m = DRestrict (Dom (Base SplitR xs)) (Base MapR m)

t15 :: SplitMap SS Int -> SplitMap SS Char -> Exp (SplitMap SS Char)
t15 xs zs = DRestrict (Dom (Base SplitR xs)) (Base SplitR zs)

t16 :: SplitMap SS Char -> Map SS Int -> Int -> Exp (SplitMap SS Char)
t16 xs m k = DRestrict (Dom (RRestrict (Base MapR m) (SetSingleton k))) (Base SplitR xs)

t17 :: SplitMap SS Char -> Map SS Int -> Map Int () -> Exp (SplitMap SS Char)
t17 xs m1 m2 = DRestrict (Dom (RRestrict (Base MapR m1) (Base MapR m2))) (Base SplitR xs)

t18 :: SplitMap SS Int -> Map SS () -> Exp (SplitMap SS Int)
t18 xs set = DRestrict (Base MapR set) (Base SplitR xs)

t19 :: SplitMap SS Int -> SS -> Exp (SplitMap SS Int)
t19 xs k = DExclude (SetSingleton k) (Base SplitR xs)

t20 :: SplitMap SS Int -> SS -> Char -> Exp (SplitMap SS Int)
t20 xs k v = DExclude (Dom (Singleton k v)) (Base SplitR xs)

t21 :: SplitMap SS Int -> Int -> SS -> Exp (SplitMap SS Int)
t21 xs k v = DExclude (Rng (Singleton k v)) (Base SplitR xs)

t22 :: SplitMap SS Int -> Set.Set SS -> Exp (SplitMap SS Int)
t22 xs set = DExclude (Base SetR (Sett set)) (Base SplitR xs)

t23 :: SplitMap SS Int -> Map SS Char -> Exp (SplitMap SS Int)
t23 xs m = DExclude (Dom (Base MapR m)) (Base SplitR xs)

t24 :: SplitMap SS Int -> SplitMap SS Char -> Exp (SplitMap SS Char)
t24 xs zs = DExclude (Dom (Base SplitR xs)) (Base SplitR zs)

t25 :: SplitMap SS Int -> Int -> Exp (SplitMap SS Int)
t25 xs u = RExclude (Base SplitR xs) (SetSingleton u)

t26 :: SplitMap SS Int -> Sett Int () -> Exp (SplitMap SS Int)
t26 xs set = RExclude (Base SplitR xs) (Base SetR set)

t27 :: SplitMap SS Int -> Int -> Exp (SplitMap SS Int)
t27 xs k = RRestrict (Base SplitR xs) (SetSingleton k)

t28 :: SplitMap SS Int -> Set.Set Int -> Exp (SplitMap SS Int)
t28 xs s = RRestrict (Base SplitR xs) (Base SetR (Sett s))
```

Reply:
FYI, I broke github when I was making this suggestion. It was giving me a 500 error, that's why this suggestion is duplicated many times, not because redundant parenthesis bother me that much :smile: